infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -294,9 +295,15 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
      * Warning: Clears all data.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
-    private void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    private synchronized void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+        // Waits for all current writes to get flushed
+        // This method is called after a seal and before consensus
+        // therefore writes wouldn't interleave with the reset
+        log.info("resetLogUnit: reset server msg1 {}", msg);
         streamLog.reset();
         dataCache.invalidateAll();
+        batchWriter = new BatchWriter(streamLog);
```

Comment:
Why do you need a new BatchWriter instance? Can we avoid this? Or clean shutdown the writerService, at least.

Reply:
hmm, I initially had batchWriter.close() then batchWriter = new BatchWriter(streamLog);
I'll add that back. The reason i create a new instance is because I close the current one. The reason I close the current one is because it waits for all the requests to get flushed.

Reply:
I think cleanly shutting down the writerService and restarting it should suffice for flushing it right (you should be guaranteed the writerService thread exits if it is running if you call cleanShutdown) --- 

but I see that
```
                    } catch (Exception e) {
                        currOp.setException(e);
                        res.add(currOp);
                    }
```
In the batchWriterProcessor() method could swallow an InterruptedException, so you need to fix that.

Reply:
Sorry, meant
```
executorService.shutdownNow();
executorService.awaitTermination();
```
was thinking about netty thread groups there